kyo-core/shared/src/main/scala/kyo/scheduler/IOPromise.scala

```diff
@@ -263,18 +263,18 @@ private[kyo] object IOPromise extends IOPromisePlatformSpecific:
 
         final def interrupts(p: IOPromise[?, ?]): Pending[E, A] =
             new Pending[E, A]:
-                def interrupt(panic: Panic): Pending[E, A] =
-                    discard(p.interrupt(panic))
+                def interrupt[E2 >: E](error: Error[E2]) =
+                    discard(p.interrupt(error.asInstanceOf[Error[E]]))
                     self
                 def waiters: Int = self.waiters + 1
                 def run[E2 >: E, A2 >: A](v: Result[E2, A2]) =
                     self
 
         @nowarn("msg=anonymous")
-        inline def onInterrupt(inline f: Panic => Unit): Pending[E, A] =
+        inline def onInterrupt(inline f: Error[E] => Unit): Pending[E, A] =
             new Pending[E, A]:
-                def interrupt(panic: Panic): Pending[E, A] =
-                    f(panic)
+                def interrupt[E2 >: E](error: Error[E2]) =
+                    f(error.asInstanceOf[Error[E]])
```

Comment:
Is there a reason the casting is necessary? Perhaps we can just coerce it with the `: Error[E]`?

Reply:
Just coercing isn't enough. I've just tried a few things: removing variance from `IOPromise` but keeping it in `Fiber`, updating the type parameters, etc but nothing worked. Variance is a nightmare

Reply:
I've created https://github.com/getkyo/kyo/issues/801